src/Functions/FunctionsHashing.h

```diff
@@ -1318,6 +1336,46 @@ class FunctionAnyHash : public IFunction
                     column->getName(), getName());
     }
 
+    template <bool first>
+    void executeNullable(const KeyType & key, const IDataType * from_type, const IColumn * column, typename     ColumnVector<ToType>::Container & vec_to) const
+    {
+        if (const auto * col_from = checkAndGetColumn<ColumnNullable>(column)) [[likely]]
+        {
+            const auto * nested_type = typeid_cast<const DataTypeNullable &>(*from_type).getNestedType().get();
+            if (nested_type->getTypeId() != TypeIndex::Nothing)
+            {
+                const auto & nested_col = col_from->getNestedColumn();
+                typename ColumnVector<ToType>::Container vec_temp(nested_col.size());
+                bool nested_is_first = true;
+                executeForArgument(key, nested_type, &nested_col, vec_temp, nested_is_first);
+                size_t j = 0;
+                for (size_t i = 0; i < vec_to.size(); ++i)
+                {
+                    ToType h{null_magic_number};
+                    if (!col_from->isNullAt(i))
+                        h = vec_temp[j++];
```

Comment:
No, j is the index of nested column.

Reply:
@aiven-sal Got it. Since there is default data in nested column if corresponding data is NULL, actually there is no need to store. Maybe we should delete the default data in another PR if it worth doing.